src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -3069,70 +3050,103 @@ def _record_exist_func(record_type):
 
 
 def _add_record(record_set, record, record_type, is_list=False):
-    record_property = _type_to_property_name(record_type)
+    record_property, _ = _type_to_property_name(record_type)
 
     if is_list:
-        record_list = getattr(record_set, record_property)
+        record_list = record_set.get(record_property, None)
         if record_list is None:
-            setattr(record_set, record_property, [])
-            record_list = getattr(record_set, record_property)
+            record_set[record_property] = []
+            record_list = []
 
         _record_exist = _record_exist_func(record_type)
         if not _record_exist(record, record_list):
-            record_list.append(record)
+            record_set[record_property].append(record)
     else:
-        setattr(record_set, record_property, record)
+        record_set[record_property] = record
 
 
 def _add_save_record(cmd, record, record_type, record_set_name, resource_group_name, zone_name,
                      is_list=True, subscription_id=None, ttl=None, if_none_match=None):
     from azure.core.exceptions import HttpResponseError
-    ncf = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_NETWORK_DNS,
-                                  subscription_id=subscription_id).record_sets
 
+    record_snake, record_camel = _type_to_property_name(record_type)
     try:
-        record_set = ncf.get(resource_group_name, zone_name, record_set_name, record_type)
+        ret = _DNSRecordSetShow(cli_ctx=cmd.cli_ctx)(command_args={
+            "name": record_set_name,
+            "zone_name": zone_name,
+            "subscription": subscription_id,
+            "resource_group": resource_group_name,
+            "record_type": record_type
+        })
+        record_set = dict()
+        record_set["ttl"] = ret.get("TTL", None)
+        record_set[record_snake] = ret.get(record_camel, None)
+        record_set = _convert_to_snake_case(record_set)
```

Comment:
Why need convert to snack case?

Reply:
I think AAZCommands can auto parse camel case argument key automatically. https://github.com/Azure/azure-cli/blob/efc8487c4fbce39ecfa000be373cb8130200c804/src/azure-cli-core/azure/cli/core/aaz/_field_type.py#L180-L183